pkgs/development/beam-modules/mix-release.nix

```diff
@@ -58,6 +58,20 @@ stdenv.mkDerivation (overridable // {
   configurePhase = attrs.configurePhase or ''
     runHook preConfigure
 
+    mkdir -p _build/$MIX_ENV/lib
+    while read -d ':' lib; do
+      for dir in $lib/*; do
+        # Strip version number for directory name if it exists, so naming of
+        # all libs matches what mix's expectation.
+        dest=$(basename $dir | cut -d '-' -f1)
+        echo "Linking $dir to _build/$MIX_ENV/lib/$dest"
+        # Symlink libs to _build so that mix can find them when compiling.
+        # This is what allows mix to compile the package without searching
+        # for dependencies over the network.
+        ln -s $dir _build/$MIX_ENV/lib/$dest
+      done
+    done <<< "$ERL_LIBS:"
```

Comment:
Not sure if we want to do something about this code duplication.

Reply:
Thank you for reviewing!

You're right here. I'm not sure how to abstract in a shell function and share with different derivations.

Reply:
We could make a hook which is run in both files.

Reply:
I think you're right here, I don't understand yet how hooks work really well. I'm going to have a go at it soon.